test/linalg.jl

```diff
@@ -26,3 +26,24 @@ for ff in (lu, svd, qr)
     @test ldiv!(FF, bbb) === bbb
     @test A*bbb ≈ b
 end
+
+# linalg mul! overloads
+n, m, l = 5, 6, 7
+bb = rand(n, n), rand(m, n), rand(l, n)
+cc = rand(n), rand(n), rand(n)
+dd = rand(n), rand(m), rand(l)
+b = ArrayPartition(bb)
+c = ArrayPartition(cc)
+d = ArrayPartition(dd)
+A = rand(n)
+for T in (Array{Float64}, Array{ComplexF64},)
+    B = T(A)
```

Comment:
Without `local` a soft scope warning is printed since it seems a global variable `B` already exists:
```suggestion
    local B = T(A)
```
The same problem exists in line 10 BTW.

Reply:
We should fix the whole test file at this point in order to avoid this tedious warnings.
Next commit I'll have polished the file.